weather_underground_base.py

```diff
@@ -40,59 +41,37 @@ class WeatherUndergroundBase(RESTPolling):
     def __init__(self):
         super().__init__()
         self._api_endpoint = 'conditions'
-        self._response_key = 'current_observation'
 
     def configure(self, context):
         super().configure(context)
 
-    def _process_response(self, resp):
-        """ Extract weather conditions from response.
+    def process_signals(self, signals):
+        response = None
+        weather_signals = []
 
-        Args:
-            resp (Response)
+        for signal in signals:
+            for locations in self.queries():
```

Comment:
Yeah. If we want multiple cities/states just use multiple blocks. This will become important when we get into retry loops and stuff like that. Keeping track of different city/state combo statuses is tricky and it's way easier to just throw another block next to it.